grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -1093,6 +1093,19 @@ class StockMovementService {
             throw new IllegalArgumentException("Could not find shipment for stock movement with ID ${stockMovement.id}")
         }
 
+        if (stockMovement.requisition.status == RequisitionStatus.ISSUED) {
+            shipment.name = stockMovement.name != shipment.name || shipment.destination == stockMovement.destination ? stockMovement.name : stockMovement.generateName()
+
+            if (shipment.destination != stockMovement.destination) {
+                shipment.outgoingTransactions?.each { transaction ->
+                    transaction.destination = stockMovement.destination
+                    transaction.save()
+                }
+            }
+        } else {
+            shipment.name = stockMovement.generateName()
```

Comment:
I don't understand the logic here. Shouldn't we change the name IFF the destination has changed?

Reply:
this is called when status is not issued, in this case we should always generate the name, because this field is not editable